### PR TITLE
fix: rename role to iam_role

### PR DESCRIPTION
**Pull request by yrobla**
When adding support to users on IAM, the model needed to be created as IamUser, because it was conflicting with existing User entity. This caused an inconsistency with Role entity for IAM. To be consistent we need to rename to IamRole.

Signed-off-by: Yolanda Robla <info@ysoft.biz>
Closes: #1344

### CODE REVIEW

**Comment by depombo** [comment]
The docs also have a few places where we need to change the name

**Comment by yrobla** [comment]
> The docs also have a few places where we need to change the name

I know but this is a different repo, right? how do you want to manage that, first we merge this then i amend the other repos? or i fix everything and i land at the same tiime?

**Comment by depombo** [comment]
> > The docs also have a few places where we need to change the name
> 
> I know but this is a different repo, right? how do you want to manage that, first we merge this then i amend the other repos? or i fix everything and i land at the same tiime?

It used to be a different repo, but now it is just in the top level docs folder. It can be in a separate / follow up PR if you prefer though.

File: docs/versioned_docs/version-0.0.18/fargate-tutorials/prisma.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ If the function call is successful, it will return a virtual table with a record
        module_name        |      created_table_name       | record_count
 --------------------------+-------------------------------+--------------
  aws_cloudwatch           | log_group                     |            0
- aws_iam                  | role                          |            0
+ aws_iam                  | iam_role                      |            0
```
Comment (depombo) [inline docs/versioned_docs/version-0.0.18/fargate-tutorials/prisma.md:58]: technically this is for an old version so it should stay the same

File: docs/versioned_docs/version-0.0.18/fargate-tutorials/flyway.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ If the function call is successful, it will return a virtual table with a record
        module_name        |      created_table_name       | record_count
 --------------------------+-------------------------------+--------------
  aws_cloudwatch           | log_group                     |            0
- aws_iam                  | role                          |            0
+ aws_iam                  | iam_role                      |            0
```
[the remark this answers is not in the document]
Comment (depombo) [reply docs/versioned_docs/version-0.0.18/fargate-tutorials/flyway.md:58]: everything within `versioned_docs` can be left alone

**Review by depombo — APPROVED** [review]
minor comments, but otherwise looks good

File: docs/versioned_docs/version-0.0.18/fargate-tutorials/flyway.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (yrobla) [reply docs/versioned_docs/version-0.0.18/fargate-tutorials/flyway.md:58]: ok let's do it in a follow up, if there is nothing more to amend,so i avoid retriggering the CI again

**Review by dfellis — APPROVED** [review]

**Review by aguillenv — APPROVED** [review]

**Pull request merged**